R/filter_panel_api.R

```diff
@@ -135,10 +69,9 @@ NULL
 #' @export
 set_filter_state <- function(datasets, filter) {
   checkmate::assert_multi_class(datasets, c("FilteredData", "FilterPanelAPI"))
-  checkmate::assert_list(filter, min.len = 0, null.ok = TRUE)
-  if (length(filter) > 0) {
-    datasets$set_filter_state(filter)
-  }
+  checkmate::assert_class(filter, "teal_slices")
```

Comment:
we still want to support the old way
```suggestion
  checkmate::assert(
    checkmate::check_class(filter, "teal_slices"),
    checkmate::check_list(filter, min.len = 0),
    checkmate::check_null(filter)
  )
```